src/interpreter/memory.rs

```diff
@@ -9,6 +9,7 @@ use std::{ops, ptr};
 
 #[derive(Debug, Clone, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[allow(clippy::derive_hash_xor_eq)] // Manual impl doesn't violate Hash guarantees
```

Comment:
of all clippy lints, this is the one I dislike the most 😂

what if we allow this as global/per default?

Reply:
I personally don't hate it, as it's a common error and I don't mind having to disable the lint by hand. Still, I'm not against disabling this globally.